code/modules/mob/living/simple_animal/hostile/mimite.dm

```diff
@@ -0,0 +1,339 @@
+	pixel_y = base_pixel_y
+	pixel_x = base_pixel_x
+	density = target.density
+	mobchatspan = initial(mobchatspan)
+
+	mimite_time = world.time + MIMITE_COOLDOWN
+	med_hud_set_health()
+	med_hud_set_status() //we're an object honest
+	vision_range = 1
+	aggro_vision_range = 1
+	return
+
+/mob/living/simple_animal/hostile/mimite/proc/restore(var/intentional = FALSE)
+	if(!morphed)
+		if(intentional)
+			to_chat(src, "<span class='warning'>You're already in your normal form!</span>")
+		return
+	morphed = FALSE
+	form = null
+	alpha = initial(alpha)
+	color = initial(color)
+	animate_movement = SLIDE_STEPS
+	maptext = null
+	density = initial(density)
+
+	visible_message("<span class='warning'>[src] suddenly collapses in on itself, turning into a strange shifting black mass!</span>", \
+					"<span class='notice'>You reform to your normal body.</span>")
+	name = initial(name)
+	icon = initial(icon)
+	icon_state = initial(icon_state)
+	vision_range = initial(vision_range)
+	aggro_vision_range = initial(aggro_vision_range)
+	cut_overlays()
+
+	//Baseline stats
+	melee_damage = initial(melee_damage)
+	set_varspeed(initial(speed))
+
+	mimite_time = world.time + MIMITE_COOLDOWN
+	med_hud_set_health()
+	med_hud_set_status() //we are not an object
+
+
+/mob/living/simple_animal/hostile/mimite/Aggro() // automated only
+	..()
+	restore()
+
+/mob/living/simple_animal/hostile/mimite/AIShouldSleep(var/list/possible_targets)
+	. = ..()
+	if(.)
+		if(!morphed)
+			var/list/things = list()
+			for(var/atom/A as() in view(src))
+				if(allowed(A))
+					things += A
+			if(LAZYLEN(things) >= 1)
+				var/atom/movable/T = pick(things)
+				assume(T)
+
+/mob/living/simple_animal/hostile/mimite/can_track(mob/living/user)
+	if(morphed)
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/mimite/AttackingTarget()
+	if(morphed)
+		restore()
+	return ..()
+
+//Ambush attack
+/mob/living/simple_animal/hostile/mimite/attack_hand(mob/living/carbon/human/M)
+	if(morphed)
+		M.Knockdown(40)
+		M.reagents.add_reagent(/datum/reagent/toxin/morphvenom/mimite, 10)
+		to_chat(M, "<span class='userdanger'>[src] bites you!</span>")
+		visible_message("<span class='danger'>[src] violently bites [M]!</span>",\
+				"<span class='userdanger'>You ambush [M]!</span>", null, COMBAT_MESSAGE_RANGE)
+		restore()
+	else
+		..()
+
+/mob/living/simple_animal/hostile/mimite/Life()
+	..()
+	if(isturf(loc) && replicate)
+		mimite_growth += rand(0,2)
+		if(mimite_growth >= 150)
+			if(!grow_as)
+				grow_as = pick(/mob/living/simple_animal/hostile/mimite, /mob/living/simple_animal/hostile/mimite/crate, /mob/living/simple_animal/hostile/mimite/ranged)
+			var/mob/living/simple_animal/hostile/mimite/S = new grow_as(src.loc)
+			playsound(S.loc, 'sound/effects/meatslap.ogg', 20, TRUE)
+			mimite_growth = 0
+	if(AIStatus == AI_STATUS_OFF || client)
+		return
+	if(venthunt)
+		if(travelling_in_vent)
+			if(isturf(loc))
+				travelling_in_vent = 0
+				entry_vent = null
+		else if(entry_vent)
```

Comment:
You need to handle the case of entry_vent getting deleted and null it in life to prevent hard-dels

Reply:
I think I did the thing